### PR TITLE
Cristi park

### PR DESCRIPTION
**Pull request by chrisssCurry**
Extended parking functionality to occupy parking spots from DB

### CODE REVIEW

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java [hunk 1]
```diff
@@ -38,9 +42,108 @@ public ParkingSpot findByIdParkingSpot(int id) throws SQLException {
...
+    public List<ParkingSpot> getFreeParkingSpotsFromDB() {
+        List<ParkingSpot> freeSpotsFromDB = new ArrayList<ParkingSpot>();
+
+        //Open connection to DB
+        DBUtil dbUtil = new DBUtil();
+        dbUtil.open();
+
+
+        //String sql = "SELECT * FROM PARKING_SPOTS_TABLE_NAME WHERE is_occupied = false;";
+        String sql = "SELECT * FROM parking.parking_spots WHERE is_occupied = false;";
```
Comment (markosk29) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:58]: variable is inside the quotation marks

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java [hunk 2]
```diff
@@ -97,4 +97,33 @@ public ParkingFloor findByIdParkingFloor(int id) throws SQLException {
...
+
+    public int getFloorCountFromDB() throws SQLException {
+        int floorCount = 0;
+
+        //Open connection to DB
+        DBUtil dbUtil = new DBUtil();
+        dbUtil.open();
+
+
+        String sql = "SELECT COUNT(id) FROM PARKING_FLOOR_TABLE_NAME WHERE id IS NOT NULL;";
```
Comment (markosk29) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java:109]: variable is inside the quotation marks

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java [hunk 3]
```diff
@@ -38,9 +42,108 @@ public ParkingSpot findByIdParkingSpot(int id) throws SQLException {
...
+
+    //Returns parking spot IDs
+    public List<Integer> getFreeParkingSpotIDsFromDB() { //add to ParkingSpotRepository
+        List<Integer> freeSpotIDsFromDB = new ArrayList<Integer>();
+
+        //Open connection to DB
+        DBUtil dbUtil = new DBUtil();
+        dbUtil.open();
+
+        String sql = "SELECT * FROM PARKING_SPOTS_TABLE_NAME WHERE is_occupied = false;";
```
Comment (markosk29) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:95]: variable is inside the quotation marks

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java [hunk 4]
```diff
@@ -38,9 +42,108 @@ public ParkingSpot findByIdParkingSpot(int id) throws SQLException {
...
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        dbUtil.close();
+        return null;
+    }
+
+
+    public boolean parkVehicleOnDBSpot(Vehicle vehicle, int vehicleID, int spotID) {
+        String sql = "UPDATE PARKING_SPOTS_TABLE_NAME SET is_occupied = true, vehicle_id = ? WHERE id= ?;";
```
Comment (markosk29) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:128]: variable is inside the quotation marks

**Review by markosk29 — APPROVED** [review]

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java [hunk 5]
```diff
@@ -38,9 +42,105 @@ public ParkingSpot findByIdParkingSpot(int id) throws SQLException {
...
         }
 
-        finally {
-            resultSet.close();
+        return null;
+    }
+
+    //Returns parking spots
+    public List<ParkingSpot> getFreeParkingSpotsFromDB() {
+        List<ParkingSpot> freeSpotsFromDB = new ArrayList<ParkingSpot>();
```
Comment (gabrielhorgos) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:50]: fromDB not necessary, the method is already in a persistence layer (in a repository class)

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java [hunk 6]
```diff
@@ -38,9 +42,105 @@ public ParkingSpot findByIdParkingSpot(int id) throws SQLException {
...
+
+    //Returns parking spot IDs
+    public List<Integer> getFreeParkingSpotIDsFromDB() { //add to ParkingSpotRepository
+        List<Integer> freeSpotIDsFromDB = new ArrayList<Integer>();
+
+        //Open connection to DB
+        DBUtil dbUtil = new DBUtil();
+        dbUtil.open();
+
+        String sql = "SELECT * FROM " + PARKING_SPOTS_TABLE_NAME + " WHERE is_occupied = false;";
```
Comment (gabrielhorgos) [inline src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:95]: Again, same question. After adding multiple parking lots, they'll share the same table for storing the parking spots. 
This seems to return the total count of parking spots ever inserted (for all the parking lots)

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java [hunk 7]
```diff
@@ -97,4 +97,33 @@ public ParkingFloor findByIdParkingFloor(int id) throws SQLException {
         }
         return spotPairs;
     }
+
+    public int getFloorCountFromDB() throws SQLException {
```
[the remark this answers is not in the document]
Comment (chrisssCurry) [reply src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java:101]: Fixed method name

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java [hunk 8]
```diff
@@ -97,4 +97,33 @@ public ParkingFloor findByIdParkingFloor(int id) throws SQLException {
...
+        int floorCount = 0;
+
+        //Open connection to DB
+        DBUtil dbUtil = new DBUtil();
+        dbUtil.open();
+
+
+        String sql = "SELECT COUNT(id) FROM " + PARKING_FLOOR_TABLE_NAME + " WHERE id IS NOT NULL;";
```
[the remark this answers is not in the document]
Comment (chrisssCurry) [reply src/main/java/com/avangarde/parkinglot/repositories/ParkingFloorRepositoryImpl.java:109]: Deprecated method, deleted

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java (hunk 5 above)
Comment (chrisssCurry) [reply src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:50]: Fixed method name

File: src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java (hunk 6 above)
Comment (chrisssCurry) [reply src/main/java/com/avangarde/parkinglot/repositories/ParkingSpotRepositoryImpl.java:95]: Modified the method

**Pull request merged**